### ISSUE TITLE
🎁 [Feature Request]: Don't append name of the struct to the states

### ISSUE THREAD

**Issue by ozgunozerk** [ozgunozerk/state-shift#40]
### What is the feature you would like to see?

Right now, we are modifying the states by prepending the name of the struct to them.

```rust
#[derive(Debug)]
#[type_state(states = (Initial, RaceSet, LevelSet, SkillSlotsSet), slots = (Initial))]
struct PlayerBuilder {
    race: Option<Race>,
    level: Option<u8>,
    skill_slots: Option<u8>,
}
```

In the background, the states are actually `PlayerBuilderInitial`, `PlayerBuilderRaceSet`, etc.

The main reason we did this, to provide maximum flexibility for multiple invocations of this macro (to prevent collisions). 

However, most of the multiple invocation problems have been resolved and addressed. See:
#16 
#20 
#22 

Recap: we moved state declaration from `impl` to `type_state`. So, multiple impl blocks on the same struct won't cause any double invocation problem. 

Also, we addressed multiple invocation problems of the macros by appending the struct name to the generic boilerplate things (sealer trait, etc.)

However, changing the names of the states was probably too much. 

### The reasons on why it was too much

1. it hurts the developer experience as shown below (the types are too long, and rust analyzer's type annotations become unreadable):
<img width="765" alt="image" src="https://github.com/user-attachments/assets/af5eecb0-750e-451f-bfd9-7de55b26d049">

2. if the users of this library wants to do more manual stuff on their struct with specific state, they probably will hit a wall, since they don't know the name of the state is altered behind the scenes by our macros.

So, preserving the states as is, has 2 benefits:
- better DevX due to much shorter type names
- better ergonomics and less surprise for those who want to do some extra tinkering to their structs and states

And only one downside:
- try to use same state names in `#[type_state]` macro, for different structs, will cause a compiler error
    - however, it feels to me this won't be the case for the majority of the users of this library
    - same states for different structs causing a problem -> this can be mentioned in the docs

In conclusion, I think the benefits outweigh the cons.
